app/uk/gov/hmrc/individualsbenefitsandcreditsapi/domains/integrationframework/IfAwards.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualsbenefitsandcreditsapi.domains.integrationframework
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{Format, JsPath, Reads}
+import play.api.libs.json.Reads.{max, min, pattern, verifying}
+
+case class IfAwards(
+    payProfCalcDate: Option[String],
+    startDate: Option[String],
+    endDate: Option[String],
+    totalEntitlement: Option[Double],
+    workTaxCredit: Option[IfWorkTaxCredit],
+    childTaxCredit: Option[IfChildTaxCredit],
+    grossTaxYearAmount: Option[Double],
+    payments: Option[IfPayments]
+)
+
+object IfAwards {
+
+  val datePattern =
+    ("^(((19|20)([2468][048]|[13579][26]|0[48])|2000)[-]02[-]29|((19|20)[0-9]{2}[-](0[469]|11)[-]" +
+      "(0[1-9]|1[0-9]|2[0-9]|30)|(19|20)[0-9]{2}[-](0[13578]|1[02])[-](0[1-9]|[12][0-9]|3[01])|(19|20)[0-9]{2}[-]02[-]" +
+      "(0[1-9]|1[0-9]|2[0-8])))$").r
+
+  val minPaymentValue = 0.0
+  val maxPaymentValue = 1000000000000000.0
+
+  def isMultipleOfPointZeroOne(value: Double): Boolean =
+    (value * 100.0) % 1 == 0
+
+  def paymentAmountValidator: Reads[Double] =
+    min[Double](minPaymentValue) andKeep max[Double](maxPaymentValue) andKeep verifying[
+      Double](isMultipleOfPointZeroOne)
+
+  implicit val awardsFormat: Format[IfAwards] = Format(
+    (
+      (JsPath \ "payProfCalcDate")
+        .readNullable[String](pattern(datePattern, "invalid date")) and
+        (JsPath \ "startDate")
+          .readNullable[String](pattern(datePattern, "invalid date")) and
+        (JsPath \ "endDate")
+          .readNullable[String](pattern(datePattern, "invalid date")) and
+        (JsPath \ "totalEntitlement")
+          .readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "workTaxCredit").readNullable[IfWorkTaxCredit] and
+        (JsPath \ "childTaxCredit").readNullable[IfChildTaxCredit] and
+        (JsPath \ "grossYearTaxAmount")
```

Comment:
Name is incorrect. 'grossTaxYearAmount' ?

Reply:
Changed